File: source/code/dsa/graphs/python/depth_first_traversal_iterative.py
def depth_first_traversal(graph, source):
    """Use stack to traverse a graph."""
    stack = [source]
    while len(stack) > 0:
        current = stack.pop()
        print(current)
        neighbors = graph[current]
        for neighbor in neighbors:
            stack.append(neighbor)


graph = {"a": ["b", "c"], "b": ["d"], "c": ["e"], "d": [], "e": ["b"], "f": ["d"]}

File: source/code/dsa/graphs/javascript/depth_first_traversal_iterative.js
function depthFirstTraversal(graph, source) {
    const stack = [source];
    while (stack.length > 0) {
        const current = stack.pop();
        console.log(current);
        for (let neighbor of graph[current]) {
            stack.push(neighbor)
        }
    }
}

const graph = {
   "a": ["b", "c"],
   "b": ["d"],
   "c": ["e"],
   "d": [],
   "e": ["b"],
   "f": ["d"]
}
